docs/development.rst

```diff
@@ -0,0 +1,151 @@
+.. _development-section:
+
+Development documentation
+=========================
+
+High-level architecture overview
+--------------------------------
+
+``Edx-enterprise`` is a pluggable Django application designed to be run within edx-platform. It is shipped with any
+edx-platform install, but can be disabled or even removed from particular Open edX instance.
+
+General direction is to decouple ``edx-enterprise`` from the rest of the edx-platform as much as possible, since
+the overall goal is to eventually make it an independent application or a plugin for a separate admin site, and not be
+part of the edx-platform LMS.
+
+Hence the following must be taken into account:
```

Comment:
In order to maintain the separation of ``edx-enterprise`` from the ``edx-platform``, follow these practices:

 * When integrating with existing ``edx-platform`` features, use REST APIs rather than Python APIs.
 * In cases where the API in question only exists as a Python API, it should be encapsulated into a module or class on the ``edx-enterprise`` side, to simplify future migration to a REST API.
 * Both REST API and Python API clients must be covered with extensive unit tests.